src/features/room/helpers/join.tsx

```diff
@@ -116,10 +125,103 @@ export default function join(
             const ok = yield* recover(
                 function* () {
                     try {
-                        const tx: Record<string, any> = yield policy.requestDelegatedJoin()
+                        if (stakingEnabled) {
```

Comment:
This looks good. I wonder, maybe we should chop it in smaller pieces and have individual transactions recoverable?